test/run_envoy_tests.sh

```diff
@@ -9,10 +9,16 @@ BINARY_DIR=$2
 # https://bazel.build/versions/master/docs/test-encyclopedia.html. In particular, TEST_SRCDIR is
 # where we expect to find the generated outputs of various scripts preparing input data (these are
 # not only the actual source files!).
-: ${TEST_TMPDIR:=/tmp/envoy_test_tmp}
-: ${TEST_SRCDIR:=/tmp/envoy_test_runfiles}
+# It is a precondition that both $TEST_TMPDIR and $TEST_SRCDIR are empty.
+if [ -z "$TEST_TMPDIR" ] || [ -z "$TEST_SRCDIR" ]
+then
+  TEST_BASE=$(mktemp -d /tmp/envoy_test.XXXXXXXX)
```

Comment:
I guess in our private test runs I could just figure out a way of injecting env vars via cmake and maybe cleaning the directories.